Source/BSGOutOfMemoryWatchdog.m

```diff
@@ -246,4 +285,17 @@ - (NSMutableDictionary *)generateCacheInfoWithConfig:(BugsnagConfiguration *)con
     return cache;
 }
 
+// Only available on iOS/tvOS
+#if BSGOOMAvailable
+- (UIApplicationState)currentAppState {
+    // Only checked outside of app extensions since sharedApplication is
+    // unavailable to extension UIKit APIs
+    if ([BSG_KSSystemInfo isRunningInAppExtension]) {
+        return YES;
+    }
+    UIApplication *app = [UIApplication performSelector:@selector(sharedApplication)];
```

Comment:
Is there a chance this call could fail if the selector needs to be specified like this?

Reply:
No. This is to avoid a check by the compiler that applicationState is not called from app extensions, which is already accounted for. There is no compile-time means of insuring that is the case.